Add Resource and Datasource for Netflow Exporter Policy (netflowExporterPol) (DCNE-154) (#1242)
CiscoDevNet/terraform-provider-aci: docs/data-sources/netflow_exporter_policy.md

# In order to regenerate this file execute `go generate` from the repository root.
# More details can be found in the [README](https://github.com/CiscoDevNet/terraform-provider-aci/blob/master/README.md).
subcategory: "Tenant Policies"
layout: "aci"
page_title: "ACI: aci_netflow_exporter_policy"
sidebar_current: "docs-aci-data-source-aci_netflow_exporter_policy"
description: |-
  Data source for Netflow Exporter Policy
---

# aci_netflow_exporter_policy #

Data source for Netflow Exporter Policy

## API Information ##

* Class: [netflowExporterPol](https://pubhub.devnetcloud.com/media/model-doc-latest/docs/app/index.html#/objects/netflowExporterPol/overview)

* Supported in ACI versions: 2.2(1k) and later.

* Distinguished Name Formats:
  - `uni/infra/exporterpol-{name}`
  - `uni/tn-{name}/exporterpol-{name}`

## GUI Information ##

* Locations:
  - `Tenants -> Policies -> NetFlow -> NetFlow Exporters`
  - `Fabric -> Access Policies -> Policies -> Interface -> NetFlow -> NetFlow Exporters`

## Example Usage ##

```hcl

data "aci_netflow_exporter_policy" "example_tenant" {
  parent_dn = aci_tenant.example.id
  name      = "netfow_exporter"
}

```

## Schema ##

### Required ###

> * `parent_dn` - (string) The distinguished name (DN) of the parent object, possible resources:

Reviewer: Is there a default for this? like infra when tenant is not provided?

  - [aci_tenant](https://registry.terraform.io/providers/CiscoDevNet/aci/latest/docs/resources/tenant) ([fvTenant](https://pubhub.devnetcloud.com/media/model-doc-latest/docs/app/index.html#/objects/fvTenant/overview))
  - The distinguished name (DN) of classes below can be used but currently there is no available resource for it:
    - [infraInfra](https://pubhub.devnetcloud.com/media/model-doc-latest/docs/app/index.html#/objects/infraInfra/overview)

* `name` (name) - (string) The name of the Netflow Exporter Policy object.

### Read-Only ###

* `id` - (string) The distinguished name (DN) of the Netflow Exporter Policy object.
* `annotation` (annotation) - (string) The annotation of the Netflow Exporter Policy object.
* `description` (descr) - (string) The description of the Netflow Exporter Policy object.

> * `dscp` (dscp) - (string) IP dscp value.

Reviewer: ```suggestion
* `dscp` (dscp) - (string) The DSCP value of the Netflow Exporter Policy object.
```

> * `dst_addr` (dstAddr) - (string) Remote node destination IP address.
> * `dst_port` (dstPort) - (string) Remote node destination port.

Reviewer: ```suggestion
* `destination_address` (dstAddr) - (string) The destination IP address of the remote node.
* `destination_port` (dstPort) - (string) The destination port of the remote node.
```

* `name_alias` (nameAlias) - (string) The name alias of the Netflow Exporter Policy object.
* `owner_key` (ownerKey) - (string) The key for enabling clients to own their data for entity correlation.
* `owner_tag` (ownerTag) - (string) A tag for enabling clients to add their own data. For example, to indicate who created this object.

> * `source_ip_type` (sourceIpType) - (string) Type of Exporter Src IP Address: Can be one of the available management IP Address for a given leaf or a custom IP Address.

Reviewer: This explanation is not really clear to me. Could you rephrase?

> * `src_addr` (srcAddr) - (string) Source IP address.

Reviewer: ```suggestion
* `source_address` (srcAddr) - (string) The source IP address.
```

> * `ver` (ver) - (string) Collector version.

Reviewer: What is this collector version? Should we elaborate a bit more?
```suggestion
* `version` (ver) - (string) Collector version.
```

---
# Documentation generated by "gen/generator.go"; DO NOT EDIT.
# In order to regenerate this file execute `go generate` from the repository root.
# More details can be found in the [README](https://github.com/CiscoDevNet/terraform-provider-aci/blob/master/README.md).
subcategory: "Tenant Policies"
layout: "aci"
page_title: "ACI: aci_netflow_exporter_policy"
sidebar_current: "docs-aci-data-source-aci_netflow_exporter_policy"
description: |-

>   Data source for ACI Netflow Exporter Policy

Reviewer: Is this decided not to change?

>   Data source for ACI Netflow Exporter Policy

Reviewer: any updates on this?

>   Data source for ACI Netflow Exporter Policy

Author: Just made that change.

* Supported in ACI versions: 2.2(1k) and later.

* Distinguished Name Formats:
  - `uni/infra/exporterpol-{name}`
  - `uni/tn-{name}/exporterpol-{name}`

## GUI Information ##

* Locations:
  - `Tenants -> Policies -> NetFlow -> NetFlow Exporters`
  - `Fabric -> Access Policies -> Policies -> Interface -> NetFlow -> NetFlow Exporters`

## Example Usage ##

```hcl

data "aci_netflow_exporter_policy" "example_tenant" {
  parent_dn = aci_tenant.example.id
  name      = "netfow_exporter"
}

```

## Schema ##

### Required ###

* `parent_dn` - (string) The distinguished name (DN) of the parent object, possible resources:
  - [aci_tenant](https://registry.terraform.io/providers/CiscoDevNet/aci/latest/docs/resources/tenant) ([fvTenant](https://pubhub.devnetcloud.com/media/model-doc-latest/docs/app/index.html#/objects/fvTenant/overview))
  - The distinguished name (DN) of classes below can be used but currently there is no available resource for it:
    - [infraInfra](https://pubhub.devnetcloud.com/media/model-doc-latest/docs/app/index.html#/objects/infraInfra/overview)

>   - Default: `uni/infra`

Reviewer 2: Did you have a discussion on whether parent_dn should be in Optional in the data source docs?

Reviewer: what is the result of this?

Reviewer: what is the result of this?

Author: Just added a new Optional Section when the parent_dn has a default value.
